gui/webdisplay/src/TWebWindowsManager.cxx

```diff
@@ -303,8 +307,6 @@ std::shared_ptr<ROOT::Experimental::TWebWindow> ROOT::Experimental::TWebWindowsM
       return nullptr;
    }
 
-   win->SetBatchMode(batch_mode || gROOT->IsWebDisplayBatch());
-
    win->SetId(++fIdCnt); // set unique ID
```

Comment:
All these should ideally be set in the constructor of `TWebWindow` or at least (if the lock scope would be too large) end up in an `Init()` (or better `TWebWindow::InitWSHandler(...)`) function. This makes sure that we don't have parts of `TWebWindow` unset, simply because one creation site forgot to also set that part.

Reply:
I do not like idea to move this into WebWindow constructor.
Here central part - connection between THttpServer and WSHandler.
And this is exactly job of TWebWindowManager.
Also window id make sense only in connection with TWebWindowManager.
All other parts initialized inside WebWindow constructor properly

Reply:
I understand that, let me try to be more explicit about what I dislike:

Looking at this code I cannot tell which parts of `win` need to be set for `win->CreateWSHandler();` to work. We need to invade the internals of `win` (setting its data members directly), in a sequence that might be super essential to keep. But I cannot tell.

This would help:
```
double timeout = gEnv->GetValue("WebGui.OperationTmout", 50.);
auto winWS = win->InitWSHandler(Instance(), ++fIdCnt, timeout);
fServer->RegisterWS(winWS);
```
Would that be acceptable?

Reply:
I got it. Done